libsawtooth/src/journal/block_validator.rs

```diff
@@ -373,13 +371,15 @@ impl BlockValidator {
         }
     }
 
-    pub fn validate_block(&self, block: &BlockPair) -> Result<(), ValidationError> {
+    pub fn validate_block(
+        &self,
+        block: &BlockPair,
+    ) -> Result<BlockValidationResult, ValidationError> {
```

Comment:
Does this overall change imply that the caller is now responsible for updating the status store?

Reply:
Dropping this change, as there is a bit of a circular dependency currently.  This seam requires a bit more design.